prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/model/nexus/impl/ModelNexus.java

```diff
@@ -309,10 +311,14 @@ private void setData(Provider provider, ModelTransaction transaction, String mod
         // Allow an update if the resource didn't exist or if the update timestamp is
         // equal to or after the one of the current value
         if (metadata == null || !metadata.getTimestamp().isAfter(timestamp)) {
-            service.eSet(resourceFeature, data);
+            EClassifier resourceType = resourceFeature.getEType();
+            if (data == null || resourceType.isInstance(data)) {
+                service.eSet(resourceFeature, data);
+            } else {
+                service.eSet(resourceFeature, EMFUtil.convertToTargetType(resourceType, data));
```

Comment:
I'm not sure about this part. The EMFUtil methode introdcued here tries to check null again and set the default value in such a case. This will never apply as the null has already been checked.